### PR TITLE
refactor(x/ecocredit): burn basket fee and remove dist keeper

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #1157 / Closes: #835 / Ref: #383

This pull request updates `MsgCreateBasket` to burn the fee rather than sending the fee to the community funding pool and as a result we no longer need the distribution keeper in the ecocredit module.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/basket/msg_create.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func (k Keeper) Create(ctx context.Context, msg *basket.MsgCreate) (*basket.MsgC
 		}
 	}
 
-	err = k.distKeeper.FundCommunityPool(sdkCtx, fee, curator)
+	err = k.bankKeeper.BurnCoins(sdkCtx, ecocredit.ModuleName, fee)
```
Comment (technicallyty) [inline x/ecocredit/server/basket/msg_create.go:39]: we need to send the fee to the ecocredit module first before burning: https://github.com/regen-network/regen-ledger/blob/2151ad1f1feb2747c571002ad67c480adaa19c60/x/ecocredit/server/basket/msg_take.go#L57-L65

Comment (ryanchristo) [reply x/ecocredit/server/basket/msg_create.go]: Nice catch. Fixed https://github.com/regen-network/regen-ledger/pull/1167/commits/8cb40073ec7b74798c29a8c2ef759679be45e870

**Review by aleem1314 — APPROVED** [review]
lgtm

**Comment by ryanchristo** [comment]
> looks good! was curious about the fee handling here vs in CreateClass. seems like if we do keep this multi-fee structure, we'd want to just satisfy one fee, and not have to satisfy all?
>
>should we address this in another PR/Issue?

Yea, I'm leaning towards satisfying one fee as well. I opened a separate issue #1165 for fee requirements.

**Review by technicallyty — APPROVED** [review]
lgtm 👍🏻

**Pull request merged**